daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -233,6 +233,38 @@ object SBuiltin {
     }
   }
 
+  final case object SBFromTextInt64 extends SBuiltin(1) {
+    private val pattern = """[+-]?\d+""".r.pattern
```

Comment:
allowing the leading `+` has the slight disadvantage that we do not have a canonical representation for `Int64` -- the same numbers can be represented in multiple ways. i'm on the fence on whether this is a good thing or not.

Reply:
If we allow leading zeros, we have the same problem.

Reply:
Yes, see my own comment below.

Reply:
I believe we should not restrict parsing to canonical representation. For instance, the final use probably do not want to drop leading "0". 
Both Haskell and java (and any other languages I can think of) parse "01" and "1" in the same way.

Furthermore, this has no impact on determinism.